src/providers/google-vertex-ai/utils.ts

```diff
@@ -190,44 +191,108 @@ export const GoogleErrorResponseTransform: (
   return undefined;
 };
 
-const getDefFromRef = (ref: string) => {
-  const refParts = ref.split('/');
-  return refParts.at(-1);
+// Extract definition key from a JSON Schema $ref string
+const getDefFromRef = (ref: string): string | null => {
+  const match = ref.match(/^#\/\$defs\/(.+)$/);
+  return match ? match[1] : null;
 };
 
-const getRefParts = (spec: Record<string, any>, ref: string) => {
-  return spec?.[ref];
+const getDefObject = (
+  defs: Record<string, any> | undefined | null,
+  key: string | null
+): any => (key && defs ? defs[key] : undefined);
+
+// Recursively expands $ref nodes in a JSON Schema object tree
+export const derefer = (
+  schema: any,
+  defs: Record<string, any> | null = null,
+  stack: Set<string> = new Set()
+): any => {
+  if (schema === null || typeof schema !== 'object') return schema;
+  if (Array.isArray(schema))
+    return schema.map((item) => derefer(item, defs, stack));
+  const node = { ...schema };
+  const activeDefs =
+    defs ?? (node.$defs as Record<string, any> | undefined) ?? null;
+  if ('$ref' in node && typeof node.$ref === 'string') {
+    const defKey = getDefFromRef(node.$ref);
+    const target = getDefObject(activeDefs, defKey);
+    if (defKey && target) {
+      if (stack.has(defKey)) return node;
+      stack.add(defKey);
+      const resolved = derefer(target, activeDefs, stack);
+      stack.delete(defKey);
+      const keys = Object.keys(node);
+      if (keys.length === 1) return resolved;
+      const { $ref: _, ...siblings } = node;
+      return derefer({ ...resolved, ...siblings }, activeDefs, stack);
```

Comment:
I'm not sure if this intended or not, why do we need to de-ref the siblings when there's a loop for every iteration that does part? L231-235(diff), am I missing something?